server/app/services/mongodb/migrator.rb

```diff
@@ -64,17 +67,27 @@ def migrations
 
     def migrate
       ensure_indexes
-      with_dlock('mongodb_migrate', 60) do
+      release_stale_lock
+      with_dlock(LOCK_NAME, LOCK_TIMEOUT) do
         migrate_without_lock
       end
     end
 
+    def release_stale_lock
+      lock = DistributedLock.where(name: LOCK_NAME).first
+      if lock && lock.created_at < (LOCK_TIMEOUT * 3).seconds.ago
+        info "releasing stale distributed lock (locked at #{lock.created_at})"
+        lock.destroy
```

Comment:
This needs to use an atomic find-and-destroy operation... The `DistributedLockCleanupJob` uses something like this, is it atomic?

```ruby
      DistributedLock.where(created_at: {:$lt => 5.minutes.ago}).destroy
```